test/e2e/e2e.go

```diff
@@ -1673,10 +1636,16 @@ var _ = ginkgo.Describe("e2e ingress traffic validation", func() {
 			ginkgo.By("Creating an external container to send the traffic from")
 			// the client uses the netexec command from the agnhost image, which is able to receive commands for poking other
 			// addresses.
-			createClusterExternalContainer(clientContainerName, agnhostImage, []string{"--network", "kind", "-P"}, []string{"netexec", "--http-port=80"})
+			primaryProviderNetwork, err := provider.Get().PrimaryNetwork()
+			framework.ExpectNoError(err, "failed to get primary network")
+			externalContainerPort := provider.Get().GetExternalContainerPort()
```

Comment:
It seems like maybe a lot of this provider could be simplified. A lot of these could just be variables (`provider.PrimaryNetwork`) that get set from `provider.init()`...

Reply:
i wanted to go through the API to be consistent with getting other networks. Also, I thought about test cases where the primary provider changes. WDYT? you feel strongly about this now? maybe we can create an issue if you do.